src/util/util.js

```diff
@@ -550,13 +550,59 @@ const Utils = {
 
     // Initialize our delete fields array.
     const modifyFields = [];
+    // Collect tagpad keys for subsequent path adjustment
+    // (tagpad submission has additional data field)
+    const tagpadComponentsKeys = [];
 
     // Iterate through all components.
     this.eachComponent(form.components, (component, path) => {
       path = `data.${path}`;
       if (component.protected) {
         debug.removeProtectedFields('Removing protected field:', component.key);
-        modifyFields.push(deleteProp(path));
```

Comment:
Rather than add ad-hoc logic to identify and capture a protected tagpad component, isn't the issue here that the `eachComponent` method is not properly iterating over a tagpad?

Reply:
I didn’t add changes to EachComponent method because it provide each component and with path to it in form. (For example for password in tag component: ‘tagpad.password’ and it looks correct)
But for removing protected data in submission we need to get path to the component’s data in submission. Usually we need just add data before form’s path, but in the case with tagpad submission looks like this:

 "data": {
    "tagpad": [
      {
        "coordinate": {
          "x": 452,
          "y": 62,
          "width": 773,
          "height": 340
        },
        "data": {
          "password": "$2a$10$rZwpkNaxk6UDh7GLv0L2jO1u.VJojMKdrGmthE/VWKTRHnX0Mo2/a"
        }
      },
      {
        "coordinate": {
          "x": 542,
          "y": 265,
          "width": 773,
          "height": 340
        },
        "data": {
          "password": "$2a$10$bm6uYgZ7GkpLAzWniJ7KSe5/qIYGJCBdStXuZSeGLwEcilKIHRKAO"
        }
      },
…
    ],
 So path to password data looks like this ‘data.tagpad[I].data.password’ So I decided to add changes to this place.

Reply:
Could we try @formio/core's `eachComponentData` method to eliminate some of this boilerplate? Just spitballing here